Runtime IndexOutOfBoundsException

I've got a runtime "index out of bounds" exception during 1a validation for a document in both PDFBOX and GF versions of verapdf-1.1.15 and i'm not sure it's because of broken font or library issue.

And unfortunately i can't just post document here because of a copyright 😔

PDFBOX report:

```
WARNING: Exception caught when validaing item
org.verapdf.core.ValidationException: Caught unexpected runtime exception during validation
	at org.verapdf.pdfa.validation.validators.BaseValidator.validate(BaseValidator.java:109)
	at org.verapdf.processor.ProcessorImpl.validate(ProcessorImpl.java:219)
	at org.verapdf.processor.ProcessorImpl.process(ProcessorImpl.java:120)
	at org.verapdf.processor.BatchFileProcessor.processItem(BatchFileProcessor.java:98)
	at org.verapdf.processor.BatchFileProcessor.processList(BatchFileProcessor.java:74)
	at org.verapdf.processor.AbstractBatchProcessor.process(AbstractBatchProcessor.java:102)
	at org.verapdf.cli.VeraPdfCliProcessor.processFiles(VeraPdfCliProcessor.java:140)
	at org.verapdf.cli.VeraPdfCliProcessor.processPaths(VeraPdfCliProcessor.java:114)
	at org.verapdf.cli.VeraPdfCli.main(VeraPdfCli.java:99)
Caused by: java.lang.IndexOutOfBoundsException: Index: 0, Size: 0
	at java.util.ArrayList.rangeCheck(ArrayList.java:653)
	at java.util.ArrayList.get(ArrayList.java:429)
	at org.apache.pdfbox.pdmodel.font.PDType1CFont.<init>(PDType1CFont.java:100)
	at org.apache.pdfbox.pdmodel.font.PDFontFactory.createFont(PDFontFactory.java:60)
	at org.apache.pdfbox.pdmodel.PDResources.getFont(PDResources.java:109)
	at org.verapdf.model.tools.resources.PDInheritableResources.getFont(PDInheritableResources.java:80)
	at org.verapdf.model.impl.pb.operator.textshow.PBOpTextShow.getFontFromResources(PBOpTextShow.java:299)
	at org.verapdf.model.impl.pb.operator.textshow.PBOpTextShow.parseFont(PBOpTextShow.java:127)
	at org.verapdf.model.impl.pb.operator.textshow.PBOpTextShow.getVeraModelFont(PBOpTextShow.java:121)
	at org.verapdf.model.factory.operator.OperatorParser.addFontAndColorSpace(OperatorParser.java:589)
	at org.verapdf.model.factory.operator.OperatorParser.parseOperator(OperatorParser.java:266)
	at org.verapdf.model.factory.operator.OperatorFactory.operatorsFromTokens(OperatorFactory.java:124)
	at org.verapdf.model.impl.pb.pd.PBoxPDContentStream.parseOperators(PBoxPDContentStream.java:100)
	at org.verapdf.model.impl.pb.pd.PBoxPDContentStream.isContainsTransparency(PBoxPDContentStream.java:87)
	at org.verapdf.model.impl.pb.pd.PBoxPDPage.parseContentStream(PBoxPDPage.java:205)
	at org.verapdf.model.impl.pb.pd.PBoxPDPage.getContentStream(PBoxPDPage.java:192)
	at org.verapdf.model.impl.pb.pd.PBoxPDPage.getLinkedObjects(PBoxPDPage.java:134)
	at org.verapdf.pdfa.validation.validators.BaseValidator.addAllLinkedObjects(BaseValidator.java:240)
	at org.verapdf.pdfa.validation.validators.BaseValidator.checkNext(BaseValidator.java:185)
	at org.verapdf.pdfa.validation.validators.BaseValidator.validate(BaseValidator.java:136)
	at org.verapdf.pdfa.validation.validators.BaseValidator.validate(BaseValidator.java:107)
	... 8 more
```

GF report:

```
WARNING: Exception caught when validaing item
org.verapdf.core.ValidationException: Caught unexpected runtime exception during validation
	at org.verapdf.pdfa.validation.validators.BaseValidator.validate(BaseValidator.java:109)
	at org.verapdf.processor.ProcessorImpl.validate(ProcessorImpl.java:219)
	at org.verapdf.processor.ProcessorImpl.process(ProcessorImpl.java:120)
	at org.verapdf.processor.BatchFileProcessor.processItem(BatchFileProcessor.java:98)
	at org.verapdf.processor.BatchFileProcessor.processList(BatchFileProcessor.java:74)
	at org.verapdf.processor.AbstractBatchProcessor.process(AbstractBatchProcessor.java:102)
	at org.verapdf.cli.VeraPdfCliProcessor.processFiles(VeraPdfCliProcessor.java:155)
	at org.verapdf.cli.VeraPdfCliProcessor.processPaths(VeraPdfCliProcessor.java:129)
	at org.verapdf.cli.VeraPdfCli.main(VeraPdfCli.java:100)
Caused by: java.lang.ArrayIndexOutOfBoundsException: Can't get object with number 0 from INDEX with 0 elements.
	at org.verapdf.pd.font.cff.CFFIndex.get(CFFIndex.java:51)
	at org.verapdf.pd.font.cff.CFFFontProgram.parseFont(CFFFontProgram.java:75)
	at org.verapdf.pd.font.opentype.OpenTypeFontProgram.parseFont(OpenTypeFontProgram.java:121)
	at org.verapdf.gf.model.impl.pd.font.GFPDType1Font.<init>(GFPDType1Font.java:63)
	at org.verapdf.gf.model.factory.fonts.FontFactory.parseFont(FontFactory.java:70)
	at org.verapdf.gf.model.impl.operator.textshow.GFOpTextShow.parseFont(GFOpTextShow.java:227)
	at org.verapdf.gf.model.impl.operator.textshow.GFOpTextShow.getVeraModelFont(GFOpTextShow.java:137)
	at org.verapdf.gf.model.factory.operators.OperatorParser.addFontAndColorSpace(OperatorParser.java:491)
	at org.verapdf.gf.model.factory.operators.OperatorParser.parseOperator(OperatorParser.java:244)
	at org.verapdf.gf.model.factory.operators.OperatorFactory.operatorsFromTokens(OperatorFactory.java:115)
	at org.verapdf.gf.model.impl.pd.GFPDContentStream.parseOperators(GFPDContentStream.java:89)
	at org.verapdf.gf.model.impl.pd.GFPDContentStream.isContainsTransparency(GFPDContentStream.java:110)
	at org.verapdf.gf.model.impl.pd.GFPDPage.parseContentStream(GFPDPage.java:199)
	at org.verapdf.gf.model.impl.pd.GFPDPage.getContentStream(GFPDPage.java:188)
	at org.verapdf.gf.model.impl.pd.GFPDPage.getLinkedObjects(GFPDPage.java:115)
	at org.verapdf.pdfa.validation.validators.BaseValidator.addAllLinkedObjects(BaseValidator.java:240)
	at org.verapdf.pdfa.validation.validators.BaseValidator.checkNext(BaseValidator.java:185)
	at org.verapdf.pdfa.validation.validators.BaseValidator.validate(BaseValidator.java:136)
	at org.verapdf.pdfa.validation.validators.BaseValidator.validate(BaseValidator.java:107)
	... 8 more
```

--- a/fontbox/src/main/java/org/apache/fontbox/cff/CFFParser.java
+++ b/fontbox/src/main/java/org/apache/fontbox/cff/CFFParser.java
@@ -110,7 +110,7 @@ public class CFFParser
         IndexData globalSubrIndex = readIndexData(input);
 
         if (topDictIndex.getCount() == 0) {
-            throw new IOException();
+            throw new IOException("Error in cff font program parsing: top DICT INDEX is empty.");
         }
         List<CFFFont> fonts = new ArrayList<CFFFont>();
         for (int i = 0; i < nameIndex.getCount(); i++)

--- a/fontbox/src/main/java/org/apache/fontbox/cff/CFFParser.java
+++ b/fontbox/src/main/java/org/apache/fontbox/cff/CFFParser.java
@@ -109,6 +109,9 @@ public class CFFParser
         stringIndex = readIndexData(input);
         IndexData globalSubrIndex = readIndexData(input);
 
+        if (topDictIndex.getCount() == 0) {
+            throw new IOException();
+        }
         List<CFFFont> fonts = new ArrayList<CFFFont>();
         for (int i = 0; i < nameIndex.getCount(); i++)
         {

--- a/pdfbox/src/main/java/org/apache/pdfbox/pdmodel/graphics/optionalcontent/PDOptionalContentProperties.java
+++ b/pdfbox/src/main/java/org/apache/pdfbox/pdmodel/graphics/optionalcontent/PDOptionalContentProperties.java
@@ -211,6 +211,9 @@ public class PDOptionalContentProperties implements COSObjectable
     public String[] getGroupNames()
     {
         COSArray ocgs = (COSArray)dict.getDictionaryObject(COSName.OCGS);
+        if (ocgs == null) {
+            return new String[] {};
+        }
         int size = ocgs.size();
         String[] groups = new String[size];
         for (int i = 0; i < size; i++)

